link/link_test.go

```diff
@@ -137,17 +127,17 @@ func testLink(t *testing.T, link Link, opts testLinkOptions) {
 			t.Fatalf("Can't pin %T: %s", link, err)
 		}
 
-		link2, err := opts.loadPinned(path, nil)
+		link2, err := LoadPinnedLink(path, nil)
 		if err != nil {
 			t.Fatalf("Can't load pinned %T: %s", link, err)
 		}
 		link2.Close()
 
-		if reflect.TypeOf(link) != reflect.TypeOf(link2) {
+		if _, raw := link.(*RawLink); !raw && reflect.TypeOf(link) != reflect.TypeOf(link2) {
```

Comment:
This type assertion checking skips all RawLink type like XDP. I think it's better to have this if statement without this logic like below:
```Go
if reflect.TypeOf(link) != reflect.TypeOf(link2) {
```
Then fix the TestRawLink test: as we know this is a cGroup type then we need casting it with appropriate type.
```Go
testLink(t, &linkCgroup{*link}, prog)
```